contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,332 @@
+        OHM = _OHM;
+        gOHM = _gOHM;
+        sOHM = _sOHM;
+        staking = _staking;
+        treasury = _treasury;
+    }
+
+    modifier isBorrower(address _borrower) {
+        if (!borrowers[_borrower].isAllowed) revert IncurDebtV1_NotBorrower(_borrower);
+        _;
+    }
+
+    /**
+     * @notice sets the maximum debt limit for the system
+     * - onlyOwner (or governance)
+     * - must be greater than or equal to existing debt
+     * @param _limit in OHM
+     */
+    function setGlobalDebtLimit(uint256 _limit) external onlyGovernor {
+        if (_limit < totalOutstandingDebt) revert IncurDebtV1_LimitBelowOutstandingDebt(_limit);
+        globalDebtLimit = _limit;
+    }
+
+    /**
+     * @notice lets a user become a borrower
+     * - onlyOwner (or governance)
+     * - user must not be borrower
+     * @param _borrower the address that will interact with contract
+     */
+    function allowBorrower(address _borrower) external onlyGovernor {
+        if (borrowers[_borrower].isAllowed) revert IncurDebtV1_AlreadyBorrower(_borrower);
+        borrowers[_borrower].isAllowed = true;
+    }
+
+    /**
+     * @notice sets the maximum debt limit for a borrower
+     * - onlyOwner (or governance)
+     * - limit must be greater than or equal to borrower's outstanding debt
+     * - limit must be less than or equal to the global debt limit
+     * @param _borrower the address that will interact with contract
+     * @param _limit borrower's debt limit in OHM
+     */
+    function setBorrowerDebtLimit(address _borrower, uint256 _limit) external onlyGovernor isBorrower(_borrower) {
+        if (_limit < borrowers[_borrower].debt) revert IncurDebtV1_AboveBorrowersDebtLimit(_limit);
+        if (_limit > globalDebtLimit) revert IncurDebtV1_AboveGlobalDebtLimit(_limit);
+
+        borrowers[_borrower].limit = uint128(_limit);
+    }
+
+    /**
+     * @notice revoke user right to borrow
+     * - onlyOwner (or governance)
+     * - user must be borrower
+     * - borrower must not have outstanding debt
+     * @param _borrower the address that will interact with contract
+     */
+    function revokeBorrower(address _borrower) external onlyGovernor isBorrower(_borrower) {
+        if (borrowers[_borrower].debt != 0) revert IncurDebtV1_BorrowerStillHasOutstandingDebet(_borrower);
+        borrowers[_borrower].isAllowed = false;
+    }
+
+    /**
+     * @notice deposits gOHM/sOHM to use as collateral
+     * - msg.sender must be a borrower
+     * - this contract must have been approved _amount
+     * @dev will unwrap and hold as sOHM in this contract
+     * @param _amount amount of gOHM/sOHM
+     * @param _token token(gOHM/sOHM) to deposit with
+     */
+    function deposit(uint256 _amount, address _token) external isBorrower(msg.sender) {
+        Borrower storage borrower = borrowers[msg.sender];
+        IERC20(_token).safeTransferFrom(msg.sender, address(this), _amount);
```

Comment:
if token is not gOHM or sOHM this method will just let them send whatever erc20 token and do nothing. We also dont have a way to recover it. Maybe just make two methods one for deposit with gOHM one for deposit with sOHM.

Reply:
Don't think there’s any need for an extra function, can just use a requirement statement to ensure `_token` is either sOHM or gOHM.